### PR TITLE
Fix Verdaccio nightly e2e tests

### PR DESCRIPTION
**Pull request by mattsoulanille**
Replace e2e's custom local publishing logic with the `publish-npm` script. This more accurately represents how we release TFJS.

Fix publish-npm not waiting for Verdaccio to start before pushing packages.

Avoid testing WebGPU on platforms other than Chrome MacOS (for now).

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

### CODE REVIEW

File: scripts/publish-npm.ts [hunk 1]
```diff
@@ -268,23 +287,19 @@ async function main() {
...
-        `Supported packages are:\n${[...ALL_PACKAGES].join('\n')}`);
+    const requestedPackages = new Set(args.packages);
+    packages = allPackages.filter(pkg => requestedPackages.has(pkg));
+
+    // Check if there are any unsupported packages requested by the user
+    const unsupportedPackages = setDifference(requestedPackages,
+                                              new Set(packages));
+    if (unsupportedPackages.size > 0) {
+      throw new Error(`Can not publish ${[...unsupportedPackages]}. `
+              + `Supported packages are:\n${[...ALL_PACKAGES].join('\n')}`);
```
Comment (mattsoulanille) [inline scripts/publish-npm.ts:302]: This was improperly parsing the CLI args. Now, it parses all remaining arguments as packages to publish.

**Review by Linchenn — APPROVED** [review]

**Comment by mattsoulanille** [comment]
[Verdaccio tests](https://console.cloud.google.com/cloud-build/builds/c28a9599-e8ab-490c-8382-ad808e327f01?project=834911136599)

**Comment by mattsoulanille** [comment]
Don't merge this yet. Nightly verdaccio is still failing.

**Comment by mattsoulanille** [comment]
[Verdaccio nightly](https://console.cloud.google.com/cloud-build/builds/279a9b63-ad40-4920-a49d-dde0f9b842a9?project=834911136599)

**Comment by mattsoulanille** [comment]
[Nightly Verdaccio Passing](https://console.cloud.google.com/cloud-build/builds/ff9c168d-453c-4458-ace8-14a7f16d7862?project=834911136599)

File: scripts/release-util.ts [hunk 2]
```diff
@@ -617,28 +621,46 @@ export function memoize<I, O>(f: (arg: I) => Promise<O>): (arg: I) => Promise<O>
...
+    throw new Error(`Verdaccio error: ${err}`);
+  });
+  serverProcess.on('disconnect', (err: unknown) => {
+    throw new Error(`Verdaccio disconnected: ${err}`);
+  });
+
+  // Kill verdaccio when node exits.
   process.on('exit', () => {serverProcess.kill();});
+
+  await ready;
```
Comment (mattsoulanille) [inline scripts/release-util.ts:663]: This change makes sure Verdaccio has started before we try to publish to it.

File: scripts/release-util.ts [hunk 3]
```diff
@@ -146,7 +149,8 @@ export const E2E_PHASE: Phase = {
   packages: ['e2e'],
   deps: [
     'tfjs', 'tfjs-backend-cpu', 'tfjs-backend-wasm', 'tfjs-backend-webgl',
-    'tfjs-converter', 'tfjs-core', 'tfjs-data', 'tfjs-layers', 'tfjs-node'
+    'tfjs-backend-webgpu', 'tfjs-converter', 'tfjs-core', 'tfjs-data',
+    'tfjs-layers', 'tfjs-node'
```
Comment (mattsoulanille) [inline scripts/release-util.ts:153]: Add tfjs webgpu to the e2e packages so it gets correctly rewritten in verdaccio tests.

**Review by Linchenn — APPROVED** [review]
LGTM, thanks!

**Pull request merged**